core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -483,11 +483,19 @@ trait WorkspaceComponent {
       filter(_.realmGroupName === realmRef.realmName.value)
     }
 
-    def listPermissionPairsForGroups(groups: Set[RawlsGroupRef]): ReadAction[Seq[WorkspacePermissionsPair]] = {
-      val query = for {
-        accessLevel <- workspaceAccessQuery if (accessLevel.groupName.inSetBind(groups.map(_.groupName.value)) && accessLevel.isRealmAcl === false)
-        workspace <- workspaceQuery if (workspace.id === accessLevel.workspaceId)
-      } yield (workspace, accessLevel)
+    def listPermissionPairsForGroups(groups: Set[RawlsGroupRef], workspaceIds: Option[Seq[String]] = None): ReadAction[Seq[WorkspacePermissionsPair]] = {
```

Comment:
I would make this 2 different functions, one with ids and the preexisting one without since that is essentially what the inside of the function is